transformer/src/getMethods.ts

```diff
@@ -1,13 +1,96 @@
-import * as ts                     from "typescript";
-import { Context }                 from "./contexts/Context";
-import { MethodDescriptionSource } from "./declarations";
-import { getDecorators }           from "./getDecorators";
-import { getTypeCall }             from "./getTypeCall";
+import * as ts                from "typescript";
+import { Context }            from "./contexts/Context";
+import {
+	MethodDescriptionSource,
+	ParameterDescriptionSource
+}                             from "./declarations";
+import { getDecorators }      from "./getDecorators";
+import { getTypeCall }        from "./getTypeCall";
 import {
 	getAccessModifier,
 	getFunctionLikeSignature,
-	getUnknownTypeCall
-}                                  from "./helpers";
+	getType,
+	getUnknownTypeCall,
+}                             from "./helpers";
+import {
+	color,
+	log,
+	LogLevel
+}                             from "./log";
+
+/**
+ * Process the signature of the method and create a parameter description for each parameter
+ *
+ * @param {ts.Signature} signature
+ * @param {Context} context
+ * @returns {Array<ParameterDescriptionSource>}
+ */
+export function getMethodParameters(signature: ts.Signature, context: Context): Array<ParameterDescriptionSource>
+{
+	const methodParameters = signature?.getParameters();
```

Comment:
Isn't that check of signature pointless? It is not nullable and you check it before you even call getMethodParameters.